code/app/wave_equation.py

```diff
@@ -186,21 +180,148 @@ def volumeIntegralFlux(u):
-        wave function at the mapped LGL nodes in all the elements.
+
+    u             : arrayfire.Array [N_LGL N_Elements 1 1]
+                    A 1-D array containing the value of the wave function at
+                    the mapped LGL nodes in the element.
     
     Returns
     -------
-    flux_integral : arrayfire.Array [N M 1 1]
+    flux_integral : arrayfire.Array [N_LGL N_Elements 1 1]
                     A 1-D array of the value of the flux integral calculated
                     for various lagrange basis functions.
     '''
     
-    dLp_xi          = gvar.dLp_xi
-    lobatto_weights = af.transpose(gvar.lobatto_weights)
-    dLp_xi_weights  = af.broadcast(utils.multiply, lobatto_weights, dLp_xi)
-    flux            = flux_x(u)
-    flux_integral   = af.blas.matmul(dLp_xi_weights, flux)
+
+    dLp_xi        = gvar.dLp_xi
+    weight_tile   = af.transpose(af.tile(gvar.lobatto_weights, 1, gvar.N_LGL))
+    dLp_xi       *= weight_tile
+    flux          = flux_x(u)
+    weight_flux   = flux
+    flux_integral = af.blas.matmul(dLp_xi, weight_flux)
     
     return flux_integral
+
+
+def lax_friedrichs_flux(u_t_n):
+    '''
+    A function which calculates the lax-friedrichs_flux :math:`f_i` using.
+    :math:`f_i = \\frac{F(u^{i + 1}_0) + F(u^i_{N_{LGL} - 1})}{2} - \frac
+                {\Delta x}{2\Delta t} (u^{i + 1}_0 - u^i_{N_{LGL} - 1})`
+    
+    Parameters
+    ----------
+    u_t_n : arrayfire.Array [N_LGL N_Elements 1 1]
+            A 2D array consisting of the amplitude of the wave at the LGL nodes
+            at each element.
+    '''
+    
+    u_iplus1_0    = af.shift(u_t_n[0, :], 0, -1)
+    u_i_N_LGL     = u_t_n[-1, :]
+    flux_iplus1_0 = flux_x(u_iplus1_0)
+    flux_i_N_LGL  = flux_x(u_i_N_LGL)
+    
+    boundary_flux = (flux_iplus1_0 + flux_i_N_LGL) / 2 \
+                        - gvar.c_lax * (u_iplus1_0 - u_i_N_LGL) / 2
+    
+    
+    return boundary_flux 
+
+
+def surface_term(u_t_n):
+    '''
+    A function which is used to calculate the surface term,
+    :math:`L_p (1) f_i - L_p (-1) f_{i - 1}`
+    using the lax_friedrichs_flux function and lagrange_basis_value
+    from gvar module.
+    
+    Parameters
+    ----------
+    u_t_n        : arrayfire.Array [N_LGL N_Elements 1 1]
+                   A 2D array consisting of the amplitude of the wave at the LGL nodes
+                   at each element.
+    Returns
+    -------
+    surface_term : arrayfire.Array [N_LGL N_Elements 1 1]
+                   The surface term represented in the form of an array,
+                   :math:`L_p (1) f_i - L_p (-1) f_{i - 1}`, where p varies from
+                   zero to :math:`N_{LGL}` and i from zero to
+                   :math:`N_{Elements}`. p varies along the rows and i along
+                   columns.
+    
+    Reference
+    ---------
+    Link to PDF describing the algorithm to obtain the surface term.
+    
+    https://cocalc.com/projects/1b7f404c-87ba-40d0-816c-2eba17466aa8/files
```

Comment:
See issue #18 and shorten the URL. Do this at all the places where such long URLs are written.